public/static/js/project_events.js

```diff
@@ -0,0 +1,116 @@
+mciModule.controller('ProjectEventsController', ['$scope','$window', 'mciProjectRestService', 'notificationService', function($scope, $window, mciProjectRestService, notificationService) {
+  $scope.load = function() {
+    $scope.userTz = $window.userTz;
+    $scope.projectId = $window.projectId;
+    $scope.Events = [];
+    // hook up incremental loading when scrolling to bottom
+    $(window).scroll(function() {
+      if ($(window).scrollTop() + $(window).height() == $(document).height()) {
+        if ($scope.nextTs !== "") {
+          $scope.getMoreEvents($scope.nextTs);
+        }
+      }
+    });
+    $scope.getMoreEvents();
+  }
+
+  $scope.getMoreEvents = function(timestamp) {
+    var successHandler = function(resp) {
+      if(resp.data == null) {
+        return;
+      }
+
+      for (var i = 0; i < resp.data.length; i ++) {
+        event = resp.data[i];
+        diff = DeepDiff(event.before, event.after);
+        event.diff = (diff) ? diff : [];
+        for (var j = 0; j < event.diff.length; j++) {
+          eventLine = event.diff[j]
+          eventLine.text = $scope.getDiffText(eventLine);
+        }
+      }
+      $scope.Events = $scope.Events.concat(resp.data);
+      $scope.nextTs = $scope.getNextTs(resp.headers().link);
+    }
+    var errorHandler = function(resp) {
+      notificationService.pushNotification("Error loading events: " + resp.data.error, "errorHeader");
+    }
+    mciProjectRestService.getEvents($scope.projectId, timestamp, 0, { success: successHandler, error: errorHandler });
+  }
+
+  $scope.getNextTs = function(pageLink) {
+    if (!pageLink) {
+      return "";
+    }
+    nextLink = pageLink.substr(1).split(">")[0];
+    if (nextLink === "") {
+      return "";
+    }
+    return getQueryParam("ts", nextLink);
+  }
+
+  $scope.getDiffText = function(diff) {
```

Comment:
Instead of creating a copy of the admin functions could you put them in a shared js file, which is used by both event types as well as the test (admin_events.test.js)? I would rename the test file since it'll be a test for this shared code